### PR TITLE
Introduce a postDeleteHandler overload to ApplicationMgtService

### PR DESCRIPTION
**Pull request by gayashanbc**
Resolves https://github.com/wso2/product-is/issues/6732

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java [hunk 1]
```diff
@@ -117,6 +117,20 @@ public boolean doPreDeleteApplication(String applicationName, String tenantDomai
     public boolean doPostDeleteApplication(String applicationName, String tenantDomain, String userName)
             throws IdentityApplicationManagementException;
 
+    /**
+     * Define any additional actions after deleting an application
+     *
+     * @param serviceProvider
+     * @param tenantDomain
+     * @param userName
+     * @return
```
Comment (DMHP) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java]: Complete the java doc comment

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java [hunk 2]
```diff
@@ -117,6 +117,20 @@ public boolean doPreDeleteApplication(String applicationName, String tenantDomai
     public boolean doPostDeleteApplication(String applicationName, String tenantDomain, String userName)
```
Comment (DMHP) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java]: Shall we marek this as deprecated?

Comment (gayashanbc) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java]: Resolved in https://github.com/wso2/carbon-identity-framework/pull/2496/commits/d281a949399c81596e58a81897fdd41f5626ecb1

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java (hunk 1 above)
Comment (gayashanbc) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java]: https://github.com/wso2/carbon-identity-framework/pull/2496/commits/d281a949399c81596e58a81897fdd41f5626ecb1

**Pull request merged**